stdlib/jms/src/main/ballerina/jms/queue_sender.bal

```diff
@@ -51,75 +43,37 @@ public type QueueSender object {
 
     extern function initQueueSender(Session session, Destination? destination = ());
 
-    # Registers the endpoint in the service.
-    # This method is not used since QueueSender is a non-service endpoint.
+    # Sends a message to the JMS provider
     #
-    # + serviceType - type descriptor of the service
-    public function register(typedesc serviceType) {
-
-    }
+    # + message - Message to be sent to the JMS provider
+    # + return - Error if unable to send the message to the queue
+    public remote extern function send(Message message) returns error?;
 
-    # Starts the consumer endpoint
-    public function start() {
-
-    }
-
-    # Returns the caller action object of the QueueSender
+    # Sends a message to a given destination of the JMS provider
     #
-    # + return - Queue sender actions
-    public function getCallerActions() returns QueueSenderActions {
-        return self.producerActions;
-    }
-
-    # Stops the consumer endpoint
-    public function stop() {
+    # + destination - Destination used for the message sender
+    # + message - Message to be sent to the JMS provider
+    # + return - Error if sending to the given destination fails
+    public remote function sendTo(Destination destination, Message message) returns error?;
+};
 
+remote function QueueSender.sendTo(Destination destination, Message message) returns error? {
+    var session = self.config.session;
+    if (session is Session) {
+        validateQueue(destination);
+        self.initQueueSender(session, destination = destination);
+    } else {
+        log:printInfo("Message producer not properly initialized for queue " + destination.destinationName);
```

Comment:
should we return an error in this case ?

Reply:
Not applicable

Reply:
Let's address this in the next iteration we need to return an error here